### PR TITLE
Add zswap Plugin

### PR DESCRIPTION
**Pull request by quotengrote**
Hi,

i have added a new Plugin for zswap metrics.

One Question: is it possible make the values axis logaritmic? I am asking because all values range from 0-99999 usually, but one is almost ever over 40 Million, so it would be nice to see the small lines better.

Wishes
mg

### CODE REVIEW

File: plugins/system/zswap [hunk 1]
```diff
@@ -0,0 +1,93 @@
+#!/bin/bash
```
Comment (sumpfralle) [inline plugins/system/zswap:1]: I think, you can use `/bin/sh` for this plugin?
(just removing the `function` keywords)

File: plugins/system/zswap [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+=head1 LICENSE
+
+GPLv3 or later
+
+SPDX-License-Identifier: GPL-3.0-or-later
+
+=head1 MAGIC MARKERS
+
+  #%# family=auto
```
Comment (sumpfralle) [inline plugins/system/zswap:30]: Please add `#%# capabilities=autoconf`.

File: plugins/system/zswap [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+  exit 0
+fi
+
+if [ "$1" = "config" ]; then
+  # setze label
+  print_labels
+  exit 0
+fi
+
+echo duplicate_entry.value $(cat "/sys/kernel/debug/zswap/duplicate_entry")
```
Comment (sumpfralle) [inline plugins/system/zswap:84]: Maybe store the base directory in a variable in order to reduce visual noise? (`SYSFS_ZSWAP_DIR=/sys/kernel/debug/zswap`)

**Review by sumpfralle — COMMENTED** [review]
Thank you for contributing this new plugin!

Please take a look at my comments below.

File: plugins/system/zswap (hunk 3 above)
Comment (quotengrote) [reply plugins/system/zswap:84]: changed

File: plugins/system/zswap (hunk 2 above)
Comment (quotengrote) [reply plugins/system/zswap:32]: changed

File: plugins/system/zswap (hunk 1 above)
Comment (quotengrote) [reply plugins/system/zswap:1]: Didn't work

Comment (sumpfralle) [reply plugins/system/zswap:1]: What exactly failed to work?
Maybe you did not add `()` after the function name? (this is optional in bash)
Maybe run `shellcheck -s dash`? Its output may be easier to understand, than the shell's syntax error messages.

We strive to keep the number of dependencies of munin plugins minimal (as long as it does not hurt maintainability). Something like `bash` renders plugins unusable on platforms like OpenWrt.
Thank you!

Comment (quotengrote) [reply plugins/system/zswap:1]: that was it...

**Comment by sumpfralle** [comment]
I merged it now. Thank you for contributing this new plugin!

**Pull request merged**